redash/tasks.py

```diff
@@ -317,3 +318,47 @@ def execute_query(self, query, data_source_id, metadata):
 @celery.task(base=BaseTask)
 def record_event(event):
     models.Event.record(event)
+
+def notify_hipchat(alert, html, new_state):
+    try:
+        hipchat_client = hipchat.HipChat(token=settings.HIPCHAT_API_TOKEN)
+        message = '[' + new_state.upper() + '] ' + alert.name + '<br />' + html
+        hipchat_client.message_room(settings.HIPCHAT_ROOM_ID, settings.NAME, message, message_format='html')
+    except:
+        logger.exception("hipchat send ERROR.")
+
+def notify_mail(alert, html, new_state, app):
+    recipients = [s.email for s in alert.subscribers()]
+    logger.debug("Notifying: %s", recipients)
+    try:
+        with app.app_context():
+            message = Message(recipients=recipients,
+                              subject="[{1}] {0}".format(alert.name, new_state.upper()),
+                              html=html)
+            mail.send(message)
+    except:
+        logger.exception("mail send ERROR.")
+
+def notify_webhook(alert, query, html, new_state):
+    http_client = httplib2.Http(timeout=30)
+    if settings.WEBHOOK_USERNAME:
+        http_client.add_credentials(settings.WEBHOOK_USERNAME, settings.WEBHOOK_PASSWORD)
+    try:
+        data = {
+            'new_state':new_state.upper(),
+            'alert_name':alert.name,
+            'alert_id':alert.id,
+            'query_id':query.id,
+            'url_base':settings.HOST
+        }
```

Comment:
We can change `to_dict` not to embed user/query objects if not asked for:

```python
def to_dict(self, full=True):
    d = {
            'id': self.id,
            'name': self.name,
            'options': self.options,
            'state': self.state,
            'last_triggered_at': self.last_triggered_at,
            'updated_at': self.updated_at,
            'created_at': self.created_at
    }

    if full:
        d['query'] = self.query.to_dict()
        d['user'] = self.user.to_dict()
    else:
        d['query_id'] = self._data['query']
        d['user_id'] = self._data['user']

    return d
```

And then in your code call it with `full=False`.